File: clients/csharp/Exceptions/LimyeDBException.cs
namespace LimyeDB.Exceptions;

/// <summary>
/// Base exception for all LimyeDB client errors.
/// </summary>
public class LimyeDBException : Exception
{
    /// <summary>
    /// The HTTP status code associated with this error, or 0 if not applicable.
    /// </summary>
    public int StatusCode { get; }

    public LimyeDBException(string message) : base(message)
    {
        StatusCode = 0;
    }

    public LimyeDBException(string message, int statusCode) : base(message)
    {
        StatusCode = statusCode;
    }

    public LimyeDBException(string message, Exception innerException) : base(message, innerException)
    {
        StatusCode = 0;
    }

    public LimyeDBException(string message, int statusCode, Exception innerException) : base(message, innerException)
    {
        StatusCode = statusCode;
    }
}

File: clients/java/src/main/java/com/limyedb/exceptions/LimyeDBException.java
package com.limyedb.exceptions;

/**
 * Base exception for all LimyeDB client errors.
 */
public class LimyeDBException extends Exception {
    private final int statusCode;

    public LimyeDBException(String message) {
        super(message);
        this.statusCode = 0;
    }

    public LimyeDBException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public LimyeDBException(String message, Throwable cause) {
        super(message, cause);
        this.statusCode = 0;
    }

    public LimyeDBException(String message, int statusCode, Throwable cause) {
        super(message, cause);
        this.statusCode = statusCode;
    }

    /**
     * Returns the HTTP status code associated with this error, or 0 if not applicable.
     */
    public int getStatusCode() {
        return statusCode;
    }
}
